metadata-ingestion/src/datahub/configuration/kafka.py

```diff
@@ -37,6 +42,16 @@ class KafkaConsumerConnectionConfig(_KafkaConnectionConfig):
         description="Extra consumer config serialized as JSON. These options will be passed into Kafka's DeserializingConsumer. See https://docs.confluent.io/platform/current/clients/confluent-kafka-python/html/index.html#deserializingconsumer and https://github.com/edenhill/librdkafka/blob/master/CONFIGURATION.md .",
     )
 
+    # MSK IAM Authentication Configuration
+    sasl_mechanism: Optional[str] = Field(
+        default=None,
+        description="SASL mechanism for authentication. Supported values: OAUTHBEARER, AWS_MSK_IAM",
+    )
+    aws_region: Optional[str] = Field(
+        default=None,
+        description="AWS region for MSK IAM authentication. Required when using OAUTHBEARER with aws_msk_iam_sasl_signer.",
```

Comment:
this might just be because I'm not familiar with the source, but it seems strange to me that `OAUTHBEARER` is tied to aws? it feels like a pretty generic name

almost feels like there's a typo and it should read "Required when using AWS_MSK_IAM as the sasl_mechanism"